cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -727,6 +732,14 @@ def no_coexist_role_policies(self, data, **kwargs):
         if self.fields_coexist(data, ["instance_role", "additional_iam_policies"], **kwargs):
             raise ValidationError("InstanceRole and AdditionalIamPolicies can not be configured together.")
 
+    @validates_schema
+    def no_coexist_instance_profile(self, data, **kwargs):
+        """Validate that instance_profile and additional_security_groups or instance_role do not co-exist."""
+        if self.fields_coexist(data, ["instance_profile", "additional_iam_policies"], **kwargs):
+            raise ValidationError("InstanceProfile and AdditionalIamPolicies can not be configured together.")
+        if self.fields_coexist(data, ["instance_profile", "instance_role"], **kwargs):
+            raise ValidationError("InstanceProfile and InstanceRole can not be configured together.")
```

Comment:
let's add unit tests for this schema validation, see as an example `test_image_schema` in the `cli/tests/pcluster/schemas/test_cluster_schema.py`

Reply:
Added